app/services/CountriesService.scala

```diff
@@ -39,6 +41,17 @@ class CountriesService @Inject() (referenceDataConnector: ReferenceDataConnector
   def getCountries()(implicit hc: HeaderCarrier): Future[SelectableList[Country]] =
     getCountries("CountryCodesFullList")
 
+  def getFilteredCountriesOfRouting(userAnswers: UserAnswers, indexToKeep: Option[Index])(implicit hc: HeaderCarrier): Future[SelectableList[Country]] = {
```

Comment:
Can `indexToKeep` be of type Index instead of `Option[Index]` ?

Reply:
Otherwise looks good. If we make countriesOfRouting a JsArray however we can use our own zipWithIndex method that uses Index rather than Int. Something like

```
  def getFilteredCountriesOfRouting(userAnswers: UserAnswers, indexToKeep: Index)(implicit hc: HeaderCarrier): Future[SelectableList[Country]] = {
    val countriesOfRouting = userAnswers.get(CountriesOfRoutingSection).getOrElse(JsArray())

    val countries: Seq[Country] = countriesOfRouting.zipWithIndex.flatMap {
      case (_, index) if indexToKeep == index => None
      case (_, index)                                            => userAnswers.get(CountryOfRoutingPage(index))
    }

    getCountries().map(_.filterNot(countries.contains))
  }
```